geoips/interfaces/module_based/readers.py

```diff
@@ -28,5 +28,30 @@ class ReadersInterface(BaseModuleInterface):
         "standard": ["metadata_only", "chans", "area_def", "self_register"]
     }
 
+    def concatenate_metadata(self, all_metadata):
+        """Merge together metadata sourced from a list of files into one dictionary.
+
+        Where the structure of the merged metadata is a nested dictionary of metadata.
+
+        Ie. (xarray_obj has no data and is merely just a container for metadata):
+        {"METADATA": {"file0_METADATA": xarray_obj, ..., "fileX_METADATA": xarray_obj}}
+
+        Parameters
+        ----------
+        all_metadata: list of xarray.Datasets
+            - The incoming metadata from X number of files
+
+        Returns
+        -------
+        concat_md: 2D dict of xarray Datasets
+            - All metadata merged into a 2D dictionary of xarray Datasets
+        """
+        md = {"METADATA": {"source_file_attributes": {}}}
+        for md_idx in range(len(all_metadata)):
+            md["METADATA"]["source_file_attributes"][f"filename_{md_idx}"] = (
+                all_metadata[md_idx]
+            )
```

Comment:
See previous comment - the intent was for the keys to be the actual filenames, not just an index.

Reply:
Gotcha. I changed this to adhere to the xarray_standards you sent.

Reply:
Maybe these are relevant, are there others?
* https://github.com/NRLMMD-GEOIPS/geoips/pull/427/commits/d4eeb2855ef29953f07f51718f39d6326079c443
* https://github.com/NRLMMD-GEOIPS/geoips/pull/427/commits/7ce0ecb8ab2382898f47a885fcb3b6dc3389ba39

Reply:
[Here](https://github.com/NRLMMD-GEOIPS/geoips/pull/427/files#r1821411519) is where it was resolved.

Reply:
Thanks!